scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,1038 @@
+    }
+
+    @Override
+    public Serializable get(String originator, TaskId taskid, String channel, String key)
+            throws InvalidChannelException {
+        return getChannel(channel).get(key);
+    }
+
+    @Override
+    public Serializable put(String originator, TaskId taskid, String channel, String key, Serializable value)
+            throws InvalidChannelException, IOException {
+        Serializable answer = getChannel(channel).put(key, value);
+        logWithContextAndPersist(taskid,
+                                 channel,
+                                 "Put " + value + ON_KEY + QUOTE + key + QUOTE + PREVIOUS_VALUE_WAS + answer + "",
+                                 Level.DEBUG);
+        return answer;
+    }
+
+    @Override
+    public Serializable remove(String originator, TaskId taskid, String channel, String key)
+            throws InvalidChannelException, IOException {
+        Serializable answer = getChannel(channel).remove(key);
+        logWithContextAndPersist(taskid,
+                                 channel,
+                                 "Removed key " + QUOTE + key + QUOTE + PREVIOUS_VALUE_WAS + answer,
+                                 Level.DEBUG);
+        return answer;
+    }
+
+    @Override
+    public void putAll(String originator, TaskId taskid, String channel,
+            Map<? extends String, ? extends Serializable> m) throws InvalidChannelException, IOException {
+        getChannel(channel).putAll(m);
+        logWithContextAndPersist(taskid, channel, "PutAll called, with new entries: " + m, Level.DEBUG);
+    }
+
+    @Override
+    public void clear(String originator, TaskId taskid, String channel) throws InvalidChannelException, IOException {
+        getChannel(channel).clear();
+        logWithContextAndPersist(taskid, channel, "Cleared channel", Level.DEBUG);
+    }
+
+    @Override
+    public Set<String> keySet(String originator, TaskId taskid, String channel) throws InvalidChannelException {
+        return getChannel(channel).keySet();
+    }
+
+    @Override
+    public Collection<Serializable> values(String originator, TaskId taskid, String channel)
+            throws InvalidChannelException {
+        return getChannel(channel).values();
+    }
+
+    @Override
+    public Set<Map.Entry<String, Serializable>> entrySet(String originator, TaskId taskid, String channel)
+            throws InvalidChannelException {
+        return getChannel(channel).entrySet();
+    }
+
+    @Override
+    public Serializable getOrDefault(String originator, TaskId taskid, String channel, String key,
+            Serializable defaultValue) throws InvalidChannelException {
+        return getChannel(channel).getOrDefault(key, defaultValue);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void forEach(String originator, TaskId taskid, String channel, String action)
+            throws CompilationException, ClosureEvaluationException, InvalidChannelException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)